### PR TITLE
012 add new role frontend

### PR DESCRIPTION
**Pull request by Meganc3379**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -63,13 +63,23 @@ app.post('/loginForm', postLoginForm);
...
+app.get('/', async (req: express.Request, res: express.Response) => {
+  res.render("home.html");
+});
+
 
-app.get('/error', getErrorMessage);
\ No newline at end of file
+app.get('/job-roles', getAllJobRoles);
+//app.get('/job-roles/:id', getSingleJobRole);
+//app.post('/job-roles',postJobRoleForm);
```
Comment (orlaghmccloskey1) [inline src/app.ts]: take away commented code

File: src/controllers/JobRoleController.ts [hunk 2]
```diff
@@ -29,4 +30,20 @@ export const getSingleJobRole = async (req: express.Request, res: express.Respon
...
+
+export const getJobRoleForm = async (req:express.Request, res:express.Response): Promise<void> => {
+    res.render('job-role-form.html');
+}
+export const postJobRoleForm = async(req: express.Request, res: express.Response): Promise<void> => {
+    try{
+        const id = await createJobRole(req.body);
+        res.redirect('/job-roles/'+ id); //changed from regular id, jobroles
+    }catch(error) {
+    //console.error("Error creating job role:", error);
```
Comment (orlaghmccloskey1) [inline src/controllers/JobRoleController.ts]: take out comment

File: src/middleware/AuthMiddleware.ts [hunk 3]
```diff
@@ -16,7 +24,8 @@ export const allowRoles = (allowedRoles: UserRole[]) => {
 
         const decodedToken: JwtToken = jwtDecode(req.session.token);
         if (!allowedRoles.includes(decodedToken.Role)) {
-            return res.status(403).send('User role not authorised for this action');
+            //return res.status(403).send('User role not authorised for this action');
```
Comment (orlaghmccloskey1) [inline src/middleware/AuthMiddleware.ts]: take out comment

File: src/models/JobRoleRequest.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export type JobRoleRequest = {
+    //jobRoleId: number,
```
Comment (orlaghmccloskey1) [inline src/models/JobRoleRequest.ts]: take out comment

File: src/services/JobRoleService.ts [hunk 5]
```diff
@@ -29,3 +31,18 @@ export const getJobRoleById = async (id: string, token: string): Promise<JobRole
...
+export const createJobRole = async(jobRole: JobRoleRequest): Promise<number> => {
+    validateJobRoleForm(jobRole);
+    try{
+
+        const response: AxiosResponse = await axios.post('http://localhost:8080/api/job-roles', jobRole);
+
+        return response.data;
+    }catch(e){
+
+        console.log(e);
```
Comment (orlaghmccloskey1) [inline src/services/JobRoleService.ts:46]: take out console

File: src/services/JobRoleService.ts [hunk 6]
```diff
@@ -29,3 +31,18 @@ export const getJobRoleById = async (id: string, token: string): Promise<JobRole
         throw new Error('failed to get JobRole information');
     }
 }
+
+export const createJobRole = async(jobRole: JobRoleRequest): Promise<number> => {
+    validateJobRoleForm(jobRole);
+    try{
+
+        const response: AxiosResponse = await axios.post('http://localhost:8080/api/job-roles', jobRole);
```
Comment (mlynch147) [inline src/services/JobRoleService.ts]: this will fail on AWS... you shouldnt need to put the http://localhost:8080 here as its specified at line 6

**Review by orlaghmccloskey1 — APPROVED** [review]

File: src/app.ts [hunk 7]
```diff
@@ -63,13 +63,21 @@ app.post('/loginForm', postLoginForm);
...
+});
+
 
-app.get('/error', getErrorMessage);
\ No newline at end of file
+app.get('/job-roles', getAllJobRoles);
+app.get('/job-role-form',allowRoles([UserRole.Admin]), getJobRoleForm);
+app.post('/job-role-form',allowRoles([UserRole.Admin]), postJobRoleForm);
+app.get('/job-roles/:id', allowRoles([UserRole.Admin, UserRole.User]), getSingleJobRole);
+app.get('/error', getErrorMessage), allowRoles([UserRole.Admin, UserRole.User]);
```
Comment (jericoatos) [inline src/app.ts]: authorisation not needed for the error message

**Review by jericoatos — APPROVED** [review]
looks good!

File: src/app.ts (hunk 7 above)
Comment (Meganc3379) [reply src/app.ts]: Fixed!

**Pull request merged**